### PR TITLE
tests: kernel: gen_isr_table: move to new ztest API

### PR DESCRIPTION
**Pull request by ghost**
Move tests/kernel/gen_isr_table/ to use new ztest API.

Signed-off-by: Guo Lixin <lixinx.guo@intel.com>

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]

**Review by enjiamai — APPROVED** [review]

**Review by carlocaione — APPROVED** [review]

**Comment by carlescufi** [comment]
Trivial, merging without maintainer approval.

**Pull request merged**
